### PR TITLE
Switch from jshint to eslint and delint using airbnb styles

### PR DESCRIPTION
**Pull request by twalker**
This swaps out `jshint` with `eslint` as a step toward updating the repository to use ES6 and targeting later versions of Node.js.

[eslint](https://eslint.org/) provides more capabilities than jshint, and supports style presets such as the popular airbnb styles.

All files have also been delinted to comply with the new styling rules. **No functional changes have been made--only syntactic and whitespace updates.** This should improve the consistency of the JS.

I recognize this is a significant stylistic change, and style is subjective, so I will not be offended if the current style is preferred over `airbnb`'s.

### To verify
Since delinting has modified a lot of files, it's best to pull the branch and view the output locally rather than within github.

1. `npm run lint` (or `grunt eslint`)
_Expected results_
Linting should pass and exit without an error code.

2. `npm test`
_Expected results_
All tests should pass.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 language: node_js
 node_js:
-- "5"
-- "6"
-- "7"
 - "8"
```
Comment (twalker) [inline .travis.yml]: Dropping support for Node v5-v7, as they do not support most ES6 language features and have a status of"End of life"
https://github.com/nodejs/Release

If versions 5-7 need to be supported, I could look at adding Babel transpilation to the project to convert from ES6 to ES5.

Comment (gstroup) [reply .travis.yml]: Dropping support for v5 is fine.  I think v6 & v7 support most of the ES6 language features though.  I'm thinking we should keep supporting those for now... everything else looks great though.  Thanks!

Comment (twalker) [reply .travis.yml]: Yes, versions 6 & 7 do support most ES6 language features. I imagine v6 will still have some active uses for awhile longer. I've adjusted the syntax to add back support for v6 and 7. 
We won't have the `...` spread operator, but most all the other language features remain.

**Comment by twalker** [comment]
@gstroup , 
I'm going to exercise my contributor permissions and merge this. Please let me know if you'd rather I not, and I'll can revert the merge.

I'm looking forward to my next PR, which add `js` support for config files. 😄

**Comment by gstroup** [comment]
@twalker Thanks for merging this in.  Sorry, I was slacking, and didn't merge it.  The change looks good.  I'm really happy you've moved ahead with more ES6 syntax.  Thanks for all your help!

**Comment by twalker** [comment]
@gstroup , no problem--it's holiday season after all!

I wanted to make use of the updated syntax during the next PR #100 , which adds support for JS config files. I used it to shrink a 3000 line `config.json` file down to a 40 line `config.js` file.  I'm most interested in having that PR reviewed & published whenever it's convenient.

**Pull request merged**